### PR TITLE
Connection Reuse is broken in a Lambda environment:

### PR DESCRIPTION
**Pull request by phong-innomizetech**
- Fix issue find metadata does not exist with webpack

### CODE REVIEW

File: src/connection/Connection.ts [hunk 1]
```diff
@@ -480,7 +480,8 @@ export class Connection {
      */
     protected findMetadata(target: Function|EntitySchema<any>|string): EntityMetadata|undefined {
         return this.entityMetadatas.find(metadata => {
-            if (metadata.target === target)
+            // @ts-ignore
```
Comment (silentroach) [inline src/connection/Connection.ts]: no

File: src/connection/Connection.ts [hunk 2]
```diff
@@ -480,7 +480,8 @@ export class Connection {
...
+            if (metadata.target.name === target.name) {
```
Comment (silentroach) [inline src/connection/Connection.ts]: `target` can be a `string`

Comment (silentroach) [inline src/connection/Connection.ts]: missing closing brackets

what is wrong with you?

**Comment by craigcartmell** [comment]
Pushed a couple of fixes to @phong-innomizetech 's original commit.

**Comment by craigcartmell** [comment]
Original monkeypatch is here which I used for reference: https://github.com/typeorm/typeorm/issues/3427#issuecomment-504134702

**Comment by Benjamin-Dobell** [comment]
@pleerock This PR seems to be responsible for a _lot_ of open issues at the moment.

It doesn't fix Webpack support, but instead breaks it. This is trivially verifiable by looking at the line:

https://github.com/typeorm/typeorm/pull/4804/files#diff-9ada25e7bb500fdf98eeb214f7c94e9aR483

What this does is compare _runtime_ names of functions, that's almost certainly going to fail in production, as production code is typically minified. Minified names are by no means guaranteed to be unique, so they shouldn't ever be used in a lookup.

In my case, I've got two classes that were both minified to `e`, so when I tried `getRepository(SomeModel)` it returns the wrong repository.

The [previous functionality](https://github.com/typeorm/typeorm/pull/4804/files#diff-9ada25e7bb500fdf98eeb214f7c94e9aL483) on the other hand was correct, in that it compared functions themselves, not their runtime names. This still occurs, but only if a match isn't found on the runtime name (which won't occur).

**Comment by Benjamin-Dobell** [comment]
Actually, even in non-minified code, there's no guarantee class (function) names are unique. They probably shouldn't be involved in the lookup at all.

**Comment by Benjamin-Dobell** [comment]
@pleerock My apologies, I see this has been discussed in #4967.

**Pull request merged**